ami/models/hifigan_generator.py

```diff
@@ -0,0 +1,175 @@
+from torch.nn import Conv1d, ConvTranspose1d
+
+
+def init_weights(m: torch.nn.Module, mean: float = 0.0, std: float = 0.01) -> None:
+    classname = m.__class__.__name__
+    if classname.find("Conv") != -1:
+        m.weight.data.normal_(mean, std)
+
+
+def get_padding(kernel_size: int, dilation: int = 1) -> int:
+    return int((kernel_size * dilation - dilation) / 2)
+
+
+class ResBlock1(torch.nn.Module):
+    def __init__(self, channels: int, kernel_size: int = 3, dilation: list[int] = [1, 3, 5]) -> None:
+        super().__init__()
+        self.convs1 = torch.nn.ModuleList(
+            [
+                torch.nn.utils.weight_norm(
+                    Conv1d(
+                        channels,
+                        channels,
+                        kernel_size,
+                        1,
+                        dilation=dilation[0],
+                        padding=get_padding(kernel_size, dilation[0]),
+                    )
+                ),
+                torch.nn.utils.weight_norm(
+                    Conv1d(
+                        channels,
+                        channels,
+                        kernel_size,
+                        1,
+                        dilation=dilation[1],
+                        padding=get_padding(kernel_size, dilation[1]),
+                    )
+                ),
+                torch.nn.utils.weight_norm(
+                    Conv1d(
+                        channels,
+                        channels,
+                        kernel_size,
+                        1,
+                        dilation=dilation[2],
+                        padding=get_padding(kernel_size, dilation[2]),
+                    )
+                ),
+            ]
+        )
+        self.convs1.apply(init_weights)
+
+        self.convs2 = torch.nn.ModuleList(
+            [
+                torch.nn.utils.weight_norm(
+                    Conv1d(channels, channels, kernel_size, 1, dilation=1, padding=get_padding(kernel_size, 1))
+                ),
+                torch.nn.utils.weight_norm(
+                    Conv1d(channels, channels, kernel_size, 1, dilation=1, padding=get_padding(kernel_size, 1))
+                ),
+                torch.nn.utils.weight_norm(
+                    Conv1d(channels, channels, kernel_size, 1, dilation=1, padding=get_padding(kernel_size, 1))
+                ),
+            ]
+        )
+        self.convs2.apply(init_weights)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        for c1, c2 in zip(self.convs1, self.convs2):
+            xt = torch.nn.functional.leaky_relu(x, 0.1)
+            xt = c1(xt)
+            xt = torch.nn.functional.leaky_relu(xt, 0.1)
+            xt = c2(xt)
+            x = xt + x
+        return x
+
+    def remove_weight_norm(self) -> None:
+        for layer in self.convs1:
+            torch.nn.utils.remove_weight_norm(layer)
+        for layer in self.convs2:
+            torch.nn.utils.remove_weight_norm(layer)
```

Comment:
https://pytorch.org/docs/2.1/generated/torch.nn.utils.parametrize.remove_parametrizations.html#torch.nn.utils.parametrize.remove_parametrizations
```suggestion
        for layer in self.convs1:
            torch.nn.utils.parametrize.remove_parametrizations(layer)
        for layer in self.convs2:
            torch.nn.utils.parametrize.remove_parametrizations(layer)
```

Reply:
https://github.com/MLShukai/ami/commit/d95b59e0287e56352e913ae1527480886d3afb44 で修正